src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -159,25 +160,16 @@ protected virtual bool ShouldIncludeNode(ExecutionContext context, IHaveDirectiv
         protected virtual void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode parent)
         {
             var fields = System.Threading.Interlocked.Exchange(ref context.ReusableFields, null);
-
-            fields = CollectFieldsFrom(context, parent.GetObjectGraphType(context.Schema)!, parent.SelectionSet!, fields);
-
             var parentType = parent.GetObjectGraphType(context.Schema)!;
+            fields = CollectFieldsFrom(context, parentType, parent.SelectionSet!, fields);
 
             var subFields = new ExecutionNode[fields.Count];
 
             int i = 0;
             foreach (var kvp in fields)
             {
-                var field = kvp.Value;
-
-                var fieldDefinition = GetFieldDefinition(context.Schema, parentType, field);
```

Comment:
`GetFieldDefinition` and `throw check` after it were moved into `CollectFieldsFrom`